Tests/SentryTests/Integrations/SentryCrash/SentryCrashIntegrationTests.swift

```diff
@@ -304,23 +299,23 @@ class SentryCrashIntegrationTests: NotificationCenterTestCase {
     private func givenCurrentSession() -> SentrySession {
         // serialize sets the timestamp
         let session = SentrySession(jsonObject: fixture.session.serialize())!
-        fixture.fileManager.storeCurrentSession(session)
+        fixture.client.fileManager.storeCurrentSession(session)
         return session
     }
     
     private func givenCrashedSession() -> SentrySession {
         let session = givenCurrentSession()
-        session.endCrashed(withTimestamp: fixture.currentDateProvider.date().addingTimeInterval(5))
+        session.endCrashed(withTimestamp: CurrentDate.date().addingTimeInterval(5))
         
         return session
     }
     
     #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
     private func givenOOMAppState() {
-        let appState = SentryAppState(releaseName: TestData.appState.releaseName, osVersion: UIDevice.current.systemVersion, vendorId: UIDevice.current.identifierForVendor?.uuidString ?? "", isDebugging: false, systemBootTimestamp: fixture.currentDateProvider.date())
+        let appState = SentryAppState(releaseName: TestData.appState.releaseName, osVersion: UIDevice.current.systemVersion, vendorId: UIDevice.current.identifierForVendor?.uuidString ?? "", isDebugging: false, systemBootTimestamp: CurrentDate.date())
```

Comment:
There was an issue where two different date providers were being used for app state, which led to a test failing because a session that was serialized and then read back into memory had a different date. I'm not sure exactly how any changes I made caused that latent bug to surface, but straightening out all the date provider usages seemed like a good idea. (This might be worth a PR on its own, actually)